plugins/slo/pkg/slo/interfaces.go

```diff
@@ -0,0 +1,64 @@
+package slo
+
+import (
+	"context"
+
+	v1 "github.com/alexandreLamarre/oslo/pkg/manifest/v1"
+	"github.com/hashicorp/go-hclog"
+	"github.com/rancher/opni/pkg/slo/shared"
+	apis "github.com/rancher/opni/plugins/slo/pkg/apis/slo"
+	sloapi "github.com/rancher/opni/plugins/slo/pkg/apis/slo"
+	"google.golang.org/protobuf/proto"
+)
+
+var datasourceToImpl map[string]SLOImpl = make(map[string]SLOImpl)
+
+func init() {
+	datasourceToImpl[shared.MonitoringDatasource] = SLOMonitoring{}
+}
+
+type SLOImpl interface {
+	New(req proto.Message, p *Plugin, ctx context.Context, lg hclog.Logger) SLOImpl
```

Comment:
This method should not be inside the interface, rather you should have two constructors e.g. `NewMonitoringStore` and `NewLoggingStore` which each return some type that implements the interface

Reply:
Removed,  and added `NewMonitoringStore`,

but added an extra interface method called `WithCurrentRequest` which sets the information needed to process the current request. 

Before resolving, I'm going to implement the entire logging interface to return grpc UNIMPLEMENTED error codes.

Reply:
Done, Actually decided to remove `NewLoggingStore`, since there is not even code that fetches the logging plugin anyways